streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -235,87 +226,99 @@
                     Type.LIST,
                     Importance.HIGH,
                     CommonClientConfigs.BOOTSTRAP_SERVERS_DOC)
-            .define(CLIENT_ID_CONFIG,
```

Comment:
I did some reordering -- I never find what I am looking for there. (1) mandatory (2) high (3) medium (4) low -- ordered alphabetically per group. But it's two commits -- the first does the content change -- the second one does the code cleanup.